unlock-app/src/components/creator/CreatorLockForm.js

```diff
@@ -131,9 +127,12 @@ const CreatorLockForm = ({ hideAction, lock, saveLock }) => {
         }
         break
       case 'maxNumberOfKeys':
-        if (value !== UNLIMITED_KEYS_COUNT && !isPositiveInteger(value)) {
+        if (!isPositiveInteger(value)) {
           return 'The number of keys needs to be greater than 0'
         }
+        if (parseInt(value, 10) < lock.outstandingKeys) {
```

Comment:
```suggestion
        if (parseInt(value, 10) <= lock.outstandingKeys) {
```

In our next contract upgrade we will authorize this to be equal to the number of keys... so let's make sure the front-end can support that please :)